### PR TITLE
 ⬆️ feat: migrate browser SQLite to sqlite-wasm

### PR DESCRIPTION
**Pull request by sboagy**
## Summary

- Replace TuneTrees’ browser SQLite engine dependency from `sql.js` to `@sqlite.org/sqlite-wasm`
- Regenerate the oosync-generated SQLite client with the new runtime shape
- Bump browser-local DB reset markers to `databaseVersion: 24` and `schemaVersion: 2.1.0-sqlite-wasm-engine`
- Remove stale `public/sql-wasm` assets and the `/sql-wasm/sql-wasm.wasm` prefetch
- Update Vite config for the new SQLite WASM package and exclude stale/prebundled oosync runtime paths
- Update SQLite raw DB typings, tests, and helpers to use the new oosync raw SQLite interface
- Fix legacy SQLite migration copy statements that relied on sql.js/double-quoted string fallback behavior
- Update E2E test seeding so private tunes satisfy enforced local genre foreign keys
- Update docs and comments that described `sql.js` as the active browser engine
- Add test/worker TypeScript path mappings to avoid duplicate `postgres` type identities with local yalc-linked oosync

## Testing

- `npm run typecheck`
- `npx vitest run tests/ src/`
- `npx vitest run tests/lib/db/install-triggers.test.ts tests/lib/db/outbox-backup.test.ts tests/lib/sync/genre-filter.test.ts tests/lib/sync/server-sync-change-log.test.ts`
- Full configured SQLite migration replay under `@sqlite.org/sqlite-wasm`
- Direct Vite production build with local placeholder env vars

## Summary by CodeRabbit

* **New Features**
  * Local storage now uses the official SQLite WASM runtime with IndexedDB persistence; adds user genre selection and a media-draft outbox; practice-queue creation when adding tunes to review.

* **Bug Fixes**
  * Improved startup/build verification and removed legacy WASM prefetch to reduce missing-runtime issues.
  * More reliable genre save/validation, tighter initial sync checks, and more robust sync trigger behavior.

* **Documentation**
  * Updated guides, troubleshooting, and workflow notes for the new local DB runtime and build steps.

* **Chores**
  * Updated migrations, tests, build verification scripts, and tooling to support the runtime swap.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -112,12 +116,11 @@
     "jodit": "^4.12.2",
     "loglevel": "^1.9.2",
     "lucide-solid": "^1.14.0",
-    "oosync": "git+https://github.com/sboagy/oosync.git#main",
+    "oosync": "git+https://github.com/sboagy/oosync.git#667-upg-sqlite-wasm",
```
[the remark this answers is not in the document]
Comment (sboagy) [reply package.json]: Done.

File: docs/DB_AND_REPLICATION_ARCHITECTURE.md [hunk 2]
```diff
@@ -237,8 +237,8 @@ setupAutoPersist(); // Saves every 5 seconds if dirty flag set
 
 **Initialization Flow:**
 ```typescript
-// 1. Load WASM module
-const SQL = await initSqlJs({ locateFile: (file) => `/sql-wasm/${file}` });
+// 1. Load the official SQLite WASM module through oosync's browser runtime
+const sqlite3 = await initSqliteWasm();
```
[the remark this answers is not in the document]
Comment (sboagy) [reply docs/DB_AND_REPLICATION_ARCHITECTURE.md:247]: Done.

File: docs/PERFORMANCE_OPTIMIZATION.md [hunk 3]
```diff
@@ -246,7 +246,7 @@ const tunes = await getTunesForUser.execute({ userId });
 **Add to index.html:**
 ```html
 <!-- Preload WASM -->
-<link rel="preload" href="/sql-wasm/sql-wasm.wasm" as="fetch" crossorigin>
+Vite-managed `@sqlite.org/sqlite-wasm` asset
```
[the remark this answers is not in the document]
Comment (sboagy) [reply docs/PERFORMANCE_OPTIMIZATION.md]: Done.

File: src/lib/db/seed-data.ts [hunk 4]
```diff
@@ -45,20 +45,25 @@ export function seedDatabase(db: SqlJsDatabase, userId: string): void {
...
-      [userId]
+    const userProfileStmt = db.prepare(
+      `SELECT id FROM user_profile WHERE id = ?`
     );
+    userProfileStmt.bind([userId]);
+    const userProfileResult = [];
+    while (userProfileStmt.step()) {
+      userProfileResult.push(userProfileStmt.get());
+    }
+    userProfileStmt.free();
```
[the remark this answers is not in the document]
Comment (sboagy) [reply src/lib/db/seed-data.ts]: Done.
